### PR TITLE
docs: improve introductory documentation [MX-1073]

### PR DESCRIPTION
**Pull request by shamrt**
Completes [MX-1073](https://rewardops.atlassian.net/browse/MX-1073)

## Description

- Reorganizes, clarifies, completes most of the JSDoc markup, including: 
  - addition of missing `@async` tags and type definitions
  - addition of wholly missing JSDocs
- Also renames some private/protected functions, removing `*InContext` suffix
- Renames the documented `orders-recipients` module to `order-recipients`
- Changes the JSDoc template to [`better-docs`](https://www.npmjs.com/package/better-docs)
- Reorganizes README, and re-adds the Usage section with example methods

## Testing

- Run `npm run build:docs` 
- Peruse (bask in the wonder of) the output: `open docs/index.html`

### CODE REVIEW

**Review by Alex-Joseph — APPROVED** [review]
LGTM, HTML _on fleek_

**Review by Alex-Joseph — COMMENTED** [review]
Working on the failing test ATM 😅

File: README.md [hunk 1]
```diff
@@ -35,29 +38,78 @@ RO.config.set('clientId', 'abcdefg1234567');
 RO.config.set('clientSecret', '9876543poiuytr');
 ```
 
-See [the SDK library documentation](https://rewardops.github.io/rewardops-sdk-node/) for all config options.
+NOTE: If your program is configured to use geographic-specific PII storage, you must also set:
+
+- `piiServerUrl`: Geographic-specific PII storage server URL.
+- `supportedLocales`: List of accepted locales for the program (RFC2616 format).
```
[the remark this answers is not in the document]
Comment (shamrt) [reply README.md:44]: Yes… 👀

[the remark this answers is not in the document]
Comment (shamrt) [reply README.md:44]: Done! 🙏

File: README.md [hunk 2]
```diff
@@ -35,29 +38,78 @@ RO.config.set('clientId', 'abcdefg1234567');
...
+myProgram.orders.get('qwerty1234' callback);
+// Post a new order for the reward with id 45231 for member 'bbdd0987'
+// NOTE: The `options` object is required and must include a `reward_id` and a `member` object
+myProgram.orders.create(
+  {
+    member: {
+      id: 'jb0987',
+      full_name: 'Jolanta Banicki',
+      email: 'jolanta.b@example.com',
+    },
```
[the remark this answers is not in the document]
Comment (shamrt) [reply README.md:105]: Is it required even for V4?

File: lib/resources/orders.js [hunk 3]
```diff
@@ -183,14 +188,21 @@ const getOrderInContext = orderContext => {
 /**
  * Higher order function for creating a `create` order function in context.
  *
+ * NOTE: If the program is configured to use geographic-specific PII storage, the SDK handles the
+ * {@link module:resources/order-recipients~storeOrderRecipient storeOrderRecipient} call under the hood
+ * by making this function an alias for {@link module:resources/order-recipients~StoreOrderRecipientFunc}.
+ *
```
[the remark this answers is not in the document]
Comment (shamrt) [reply lib/resources/orders.js:194]: Bold may not be a bad idea…

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Alex-Joseph) [reply README.md:105]: Ah you are correct, it is **not** required

File: lib/resources/orders.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (shamrt) [reply lib/resources/orders.js]: Done

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (shamrt) [reply README.md:105]: Okay, we confirmed that for V4 it is _not_ required, so omitting for now.

**Comment by shamrt** [comment]
> Working on the failing test ATM 😅
> 
> UPDATE: Passing tests #37

While this PR is focused on docs, it has a few code touch-ups as well, so I'll wait until #37 is merged

**Pull request merged**